aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1452,115 +1465,114 @@ Tensor matmul(
     const Tensor& tensor1,
     const Tensor& tensor2) {
   NoNamesGuard guard;
-  auto dim_tensor1 = tensor1.dim();
-  auto dim_tensor2 = tensor2.dim();
-  auto has_out = out_opt.has_value();
-  Tensor out = out_opt.value_or(Tensor());
+  const auto dim_tensor1 = tensor1.dim();
+  const auto dim_tensor2 = tensor2.dim();
+  const auto has_out = out_opt.has_value();
 
   if (dim_tensor1 == 1 && dim_tensor2 == 1) {
-    return has_out ? at::native::dot_out(tensor1, tensor2, out) : tensor1.dot(tensor2);
+    return has_out ? at::dot_out(tensor1, tensor2, *out_opt) : tensor1.dot(tensor2);
```

Comment:
I'm confused about this (and the other `_out` below). Isn't this function a composite implicit autograd?

Reply:
Yes, but note that if out is defined, then it calls `dot_out`, otherwise it does the dispatch. @ngimel told me to remove the `native::`, but I really don't see why this should be removed in this path.

Reply:
(at least not from an autograd perspective. I have no idea about tracing and all that...)

Reply:
Ho that's only when matmul is called with `out` ok!
sounds great!